classes/app.c

```diff
@@ -168,6 +166,9 @@ PHP_MINIT_FUNCTION(UI_App)
 	uiApp_ce = zend_register_internal_class(&ce);
 	uiApp_ce->create_object = php_ui_app_create;
 
+	zend_declare_class_constant_long(uiApp_ce, ZEND_STRL("RUN_LOOP"), PHP_UI_APP_RUN_LOOP);
```

Comment:
ui is using non-conventional "Naming" for constants, we need to stick to that  ...

But not sure what they should be, here's a suggestion:

  - App::run() should just do uiMain()
  - App::run(App::Loop) should do uiMainStep(0) - so that this will be the non-blocking, no-waiting option
  - App::run(App::Loop & App::Wait) should do uiMainStep(1) - so that this will be the looping and blocking version

How about that ?